app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,26 @@
+class SessionsController < ApplicationController
+
+  def new
+  end
+
+  def create
+    user = User.find_by(email: params[:email])
+    if user && user.authenticate(params[:password]) && user.verified
+      session[:user_id] = user.id
+      if params[:remember_me]
+        user.generate_token(:remember_me_token)
+        user.save!
+        cookies.permanent[:auth_token] = user.remember_me_token
+      end
```

Comment:
if params[:remember_me]
        user.generate_token(:remember_me_token)
        user.save!
        cookies.permanent[:auth_token] = user.remember_me_token
      end

can be written as


      if params[:remember_me] && user.generate_token(:remember_me_token) &&  user.save!
        
        cookies.permanent[:auth_token] = user.remember_me_token
      en